packages/jupyter-ai/src/inline-completions.ts

```diff
@@ -0,0 +1,500 @@
+          title: 'Maximum suffix length',
+          minimum: 0,
+          type: 'number',
+          description:
+            'At most how many suffix characters should be provided to the model.'
+        },
+        disabledLanguages: {
+          title: 'Disabled languages',
+          type: 'array',
+          items: {
+            type: 'string',
+            oneOf: knownLanguages.map(language => {
+              return { const: language.name, title: displayName(language) };
+            })
+          },
+          description:
+            'Languages for which the completions should not be shown.'
+        }
+      },
+      default: JupyterAIInlineProvider.DEFAULT_SETTINGS as any
+    };
+  }
+
+  async configure(settings: { [property: string]: JSONValue }): Promise<void> {
+    this._settings = settings as unknown as JupyterAIInlineProvider.ISettings;
+  }
+
+  isEnabled(): boolean {
+    return this._settings.enabled;
+  }
+
+  isLanguageEnabled(language: string) {
+    return !this._settings.disabledLanguages.includes(language);
+  }
+
+  /**
+   * Extract prefix from request, accounting for context window limit.
+   */
+  private _prefixFromRequest(request: CompletionHandler.IRequest): string {
+    const textBefore = request.text.slice(0, request.offset);
+    const prefix = textBefore.slice(
+      -Math.min(this._settings.maxPrefix, textBefore.length)
+    );
+    return prefix;
+  }
+
+  /**
+   * Extract suffix from request, accounting for context window limit.
+   */
+  private _suffixFromRequest(request: CompletionHandler.IRequest): string {
+    const textAfter = request.text.slice(request.offset);
+    const prefix = textAfter.slice(
+      0,
+      Math.min(this._settings.maxPrefix, textAfter.length)
+    );
+    return prefix;
+  }
+
+  private _resolveLanguage(language: IEditorLanguage | null) {
+    if (!language) {
+      return 'plain English';
+    }
+    if (language.name === 'ipythongfm') {
+      return 'markdown';
+    }
+    return language.name;
+  }
+
+  private _settings: JupyterAIInlineProvider.ISettings =
+    JupyterAIInlineProvider.DEFAULT_SETTINGS;
+
+  private _currentModel = '';
+  private _counter = 0;
+}
+
+namespace JupyterAIInlineProvider {
+  export interface IOptions {
+    completionHandler: CompletionWebsocketHandler;
+    languageRegistry: IEditorLanguageRegistry;
+  }
+  export interface ISettings {
+    maxPrefix: number;
+    maxSuffix: number;
+    debouncerDelay: number;
+    enabled: boolean;
+    disabledLanguages: string[];
+  }
+  export const DEFAULT_SETTINGS: ISettings = {
+    maxPrefix: 10000,
+    maxSuffix: 10000,
+    // The debouncer delay handling is implemented upstream in JupyterLab;
+    // here we just increase the default from 0, as compared to kernel history
+    // the external AI models may have a token cost associated.
+    debouncerDelay: 250,
+    enabled: true,
+    // ipythongfm means "IPython GitHub Flavoured Markdown"
+    disabledLanguages: ['ipythongfm']
```

Comment:
We could support markdown but this would require a separate prompt template on the backend. I would suggest that this is deferred to a separate PR as I think code suggestions are higher priority.

Reply:
Agreed! Python first makes total sense to me.